app/assets/src/components/common/MetadataCSVLocationsMenu.jsx

```diff
@@ -0,0 +1,192 @@
+    }
+  });
+  return { newMetadata, warnings: allWarnings };
+};
+
+const isRowHuman = row => row["Host Genome"] && row["Host Genome"] === "Human";
+
+const NAME_COLUMN = "Sample Name";
+
+class MetadataCSVLocationsMenu extends React.Component {
+  state = {
+    // Which sample the "Apply to All" button should appear on.
+    applyToAllSample: null,
+  };
+
+  renderApplyToAll = sample => {
+    const { applyToAllSample } = this.state;
+    return applyToAllSample === sample ? (
+      <div
+        className={cs.applyToAll}
+        onClick={withAnalytics(
+          () => this.applyToAll(sample),
+          "MetadataCsvLocationsMenu_apply-all_clicked",
+          {
+            sampleName: sample.name,
+          }
+        )}
+      >
+        Apply to All
+      </div>
+    ) : null;
+  };
+
+  applyToAll = sample => {
+    const {
+      metadata,
+      metadataType,
+      onMetadataChange,
+      onCSVLocationWarningsChange,
+    } = this.props;
+
+    const newValue = (find({ [NAME_COLUMN]: sample }, metadata.rows) || {})[
+      metadataType.name
+    ];
+    const newMetadata = metadata;
+    const allWarnings = {};
+
+    // Set all the rows to the newValue
+    newMetadata.rows.forEach(row => {
+      const { result, warning } = processLocationSelection(
+        newValue,
+        isRowHuman(row)
+      );
+      row[metadataType.name] = result;
+      if (warning) allWarnings[row[NAME_COLUMN]] = warning;
+    });
+
+    onMetadataChange({
+      metadata: newMetadata,
+    });
+    onCSVLocationWarningsChange(allWarnings);
+    this.setState({ applyToAllSample: null });
+  };
+
+  // Populate the data table cells
+  getManualInputData = () => {
+    const {
+      CSVLocationWarnings,
+      metadata,
+      metadataType,
+      onMetadataChange,
+    } = this.props;
+
+    return metadata.rows.map((row, rowIndex) => {
+      const sampleName = row[NAME_COLUMN];
+
+      const onChange = (key, value) => {
+        const newMetadata = metadata;
+        newMetadata.rows[rowIndex][metadataType.name] = value;
+
+        onMetadataChange({
+          metadata: newMetadata,
+        });
+        this.setState({ applyToAllSample: sampleName });
+        logAnalyticsEvent("MetadataManualInput_input_changed", {
+          key,
+          value,
+          sampleName,
+        });
+      };
+
+      return {
+        [NAME_COLUMN]: (
+          <div className={cs.sampleName} key={NAME_COLUMN}>
+            {sampleName}
+          </div>
+        ),
+        [metadataType.key]: (
+          <div>
+            <MetadataInput
+              key={metadataType.key}
+              className={cs.input}
+              value={row[metadataType.name]}
+              metadataType={metadataType}
+              onChange={onChange}
+              withinModal={true}
+              isHuman={isRowHuman(row)}
+              warning={CSVLocationWarnings[sampleName]}
+            />
+            {this.renderApplyToAll(sampleName)}
```

Comment:
I suggest making the condition here `applyToAllSample && this.renderApplyToAll(sampleName)` instead of inside the function. More clear when you are reading the outside component and the function `renderApplyToAll` actually renders `ApplyToAll`.

Reply:
👍 done